server/src/main.py

```diff
@@ -80,10 +74,10 @@ async def stat(workspace_id: str, path: Optional[str] = "") -> list[File]:
             )
 
     try:
-        return [
-            File.from_minio_object(obj)
-            for obj in client.list_objects(workspace_id, prefix=path)
-        ]
+        objects = list(client.list_objects(workspace_id, prefix=path))
+        directories = [File.from_minio_object(obj) for obj in objects if obj.is_dir]
+        files = [File.from_minio_object(obj) for obj in objects if not obj.is_dir]
```

Comment:
I think this might be redundant, we should be able to just return `[File.from_minio_object(obj) for obj in objects]` since we are just concatenating the two.

If you've done this to order the directories before the files (and maybe it was in the AC), I think we can just do the sorting client side in hindsight.